sno/crs_util.py

```diff
@@ -50,3 +50,23 @@ def get_identifier_int(crs):
         return int(auth_code)
     # Stable code that fits easily in an int32 and won't collide with EPSG codes.
     return (hash(crs.ExportToWkt()) & 0xFFFFFFF) + 1000000
+
+
+def get_identifier_int_from_dataset(dataset, crs_name=None):
+    """
+    Get the CRS attached to this dataset with a particular name eg "EPSG:2193",
+    and return an integer to uniquely identify it, eg 2193.
+    (This still works even if the CRS is custom and doesn't have an obvious number embedded in it).
+    crs_name can be ommitted if there is only one geometry column.
+    """
+
+    if crs_name is None:
+        geom_columns = dataset.schema.geometry_columns
+        if len(geom_columns) == 1:
+            crs_name = geom_columns[0].extra_type_info.get("geometryCRS", None)
```

Comment:
what happens if we have >1 geometry column?

Reply:
Fixed this to raise ValueErrors for >1 of geometry columns. I think much of sno doesn't support having more than 1 geometry column - the new model does, in theory, but GPKG does not and any code older than V2 does not.

Reply:
sure, I think it'll be a limitation for quite a while.